src/components/RefinementList/RefinementList.js

```diff
@@ -121,20 +123,44 @@ class RefinementList extends React.Component {
     this.setState({isShowMoreOpen});
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (this.searchbox && !nextProps.isFromSearch) {
+      this.searchbox.clearInput();
+    }
+  }
+
+  refineFirstValue() {
+    const firstValue = this.props.facetValues[0];
+    if (firstValue) {
+      const actualValue = firstValue[this.props.attributeNameKey];
+      this.props.toggleRefinement(actualValue);
+    }
+  }
+
   render() {
+    const {
+      cssClasses,
+      limitMin,
+      limitMax,
+      facetValues,
+      searchFacetValues,
+      isFromSearch,
+      searchPlaceholder,
+    } = this.props;
```

Comment:
I actually never feel the struggle of repeating this.props., I like it it's descriptive about where did we find this data